### ISSUE TITLE
Getting reads from blast output

### ISSUE THREAD

**Issue by rotifergirl** [clemgoub/dnaPipeTE#11]
Hi, 

I was wondering if there is any way to match the blast output to the reads in my .fq files. I looked at the sorted and unsorted "blast.out" files in the blast_out folder, but I can't find any read names in there. I'd like to create a library of reads that fall into the "single or low copy DNA" category on the piechart so that I can assemble these regions separately. 

Any suggestions would be appreciated!

**Comment by clemgoub** [comment]
Hi!

Sorry, I was travelling!
You're right the reads are renamed... I will try to find a fix for that and let you know as soon as possible!

Cheers,

Clément

**Comment by clemgoub** [comment]
Okay here is what you can do:

The reads are renamed, but the blast sample, which contains all the read used for the 3 blast step will be found in the file: "renamed.blasting_reads.fasta"

In the blast.out file your queries are only named by numbers which are actually the read names in the "renamed.blasting_reads.fasta" file. We did it in case some input may have too long or complex/problematic names that may compromise the different steps.

Let me know if you can find them and if this makes sense!

Cheers,

Clément

**Comment by rotifergirl** [comment]
Hi Clement,

That does seem to answer my question, but now I realise that my question was not the right way to get what I want. I want to be able to eliminate the repeat sequences from my entire read library, not just the BLAST sample. I will look for a different way to subset the reads I want, but thank you for your response!

Julie

**Comment by clemgoub** [comment]
Ho yes sorry!

A quick way would be to map your whole .fq file against the Trinity.fasta output, using bwa for instance. You can then keep only the non-mapping read using the flag 4

```
samtools view -b -f 4 file.bam > file_unmapped.bam
bamToFastq -bam file_unmapped.bam -fq1 unmappedR1.fastq -fq2 unmappedR2.fastq
```

However this may discard reads that may belongs to old/highly derived copies since bwa is a classic genome mapper. Thus you can use blastn with a shorter seed for use with small reads as we do in dnaPipeTE. This will be longer but more sensitive. You can then sort your blast output and keep only the reads that have no hit on the Trinity.fasta below is a multicore (using parallel, available in the dnaPipeTE package) solution for the blast that may save you time (your reads need to be converted in fasta) that you can try:

```
### blast
cat reads.fasta | parallel -j <CPU> --block 100k --recstart '>' --pipe blastn -outfmt 6 -task dc-megablast -db Trinity.fasta -query - > reads_vs_Trinity.fasta.blast.out

### sort the blast.out and make list of reads that match a repeat
sort -k1,1 -k12,12nr -k11,11n | sort -u -k1,1 | awk '{print $1}' > matching_repeats.reads_vs_Trinity.fasta.blast.out 

### extract the reads which DO NOT match the repeat:
perl -ne 'if(/^>(\S+)/){$c=!$i{$1}}$c?print:chomp;$i{$_}=1 if @ARGV' matching_repeats.reads_vs_Trinity.fasta.blast.out reads.fasta > single_or_low_copy_DNA_reads.fasta
```

Let me know how it goes for you!

Cheers,

Clément

**Comment by rotifergirl** [comment]
Hi Clement,

I finally got back to this work, and I took a slightly different approach. I had previously already checked if the dnapipete contigs were present in my whole genome assemblies (most of them were), so I mapped my reads back to the contigs in my genome assemblies that matched my dnapipete contigs. Then I extracted the reads that didn't map to any of these repeat contigs, and it seems to be a fairly thorough method of excluding repeat reads from my library. Thanks also for your suggestion!

Julie
